### PR TITLE
[MXNet-1348][WIP][Fit API]Adding CNN examples for fit() API

### PR DESCRIPTION
**Pull request by abhinavs95**
## Description ##
Adding AlexNet, ResNet and FCN examples for the Gluon fit() API.
This PR depends on its parent PR #14346 
JIRA epic: https://issues.apache.org/jira/projects/MXNET/issues/MXNET-1333

Using the fit() API reduces the number of lines of code in each example from ~40 to ~10

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-1348](https://issues.apache.org/jira/projects/MXNET/issues/MXNET-1348)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- Added AlexNet, ResNet and FCN examples for fit() API

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [Gluon, pr-work-in-progress]

File: example/gluon/estimator_example/alexnet.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    parser.add_argument('--input-size', type=int, default=224,
+                        help='size of the input image size. default is 224')
+    parser.add_argument('--lr', type=float, default=0.001,
+                        help='learning rate. default is 0.001')
+    parser.add_argument('-j', '--num-workers', default=None, type=int,
+                        help='number of preprocessing workers')
+    opt = parser.parse_args()
+    return opt
+
+class AlexNet(HybridBlock):
```
Comment (nswamy) [inline example/gluon/estimator_example/alexnet.py]: you can fetch this from gluon-cv

**Comment by roywei** [comment]
Hi @szha, thanks for pointing out, we do plan to do a tutorial for the API. after some discussions, we dediced to move these examples to integration tests first. We will import what's available in vision model zoo, but will create these models again for RNN use cases.
Please see out updated test plan [here](https://cwiki.apache.org/confluence/display/MXNET/Gluon+Fit+API+-+Tech+Design#GluonFitAPI-TechDesign-TestPlan ). two tests for training accuracy and two teststo run all modesl (one forward and backward)

File: example/gluon/estimator_example/alexnet.py (hunk 1 above)
Comment (abhinavs95) [reply example/gluon/estimator_example/alexnet.py]: Yes. Since the gluon model zoo already has alexnet and resnet, I will import it from there.

Comment (ankkhedia) [reply example/gluon/estimator_example/alexnet.py]: @abhinavs95  You can also use default arguments  eg. learning rate , num_workers to make the example simple for the beginners

Comment (abhinavs95) [reply example/gluon/estimator_example/alexnet.py]: will do, thanks!

File: example/gluon/estimator_example/alexnet.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        transformer += [data.vision.transforms.Resize(resize)]
+    transformer += [data.vision.transforms.ToTensor()]
+    transformer = data.vision.transforms.Compose(transformer)
+    mnist_train = data.vision.MNIST(root=root, train=True)
+    mnist_test = data.vision.MNIST(root=root, train=False)
+
+    if num_workers is None:
+        num_workers = 0 if sys.platform.startswith('win32') else 4
+
+    train_iter = data.DataLoader(
```
Comment (piyushghai) [inline example/gluon/estimator_example/alexnet.py]: You can also directly load the data using a single line : 
```gluon.data.DataLoader(gluon.data.vision.FashionMNIST(train=True, transform=transform), batch_size=128, shuffle=True)```
